pkg/persistentvolumeclaim/jiva.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020-2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package persistentvolumeclaim
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/openebs/openebsctl/pkg/client"
+	"github.com/openebs/openebsctl/pkg/util"
+	"github.com/openebs/openebsctl/pkg/volume"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+const (
+	jivaPvcInfoTemplate = `
+{{.Name}} Details  :
+-------------------
+NAME               : {{.Name}}
+NAMESPACE          : {{.Namespace}}
+CAS TYPE           : {{.CasType}}
+BOUND VOLUME       : {{.BoundVolume}}
+ATTACHED TO NODE   : {{.AttachedToNode}}
+JIVA VOLUME POLICY : {{.JVP}}
+STORAGE CLASS      : {{.StorageClassName}}
+SIZE               : {{.Size}}
+JV STATUS          : {{.JVStatus}}
+PV STATUS          : {{.PVStatus}}
+`
+)
+
+// DescribeJivaVolumeClaim describes a jiva storage engine PersistentVolumeClaim
+func DescribeJivaVolumeClaim(c *client.K8sClient, pvc *corev1.PersistentVolumeClaim, vol *corev1.PersistentVolume) error {
+	// 1. Get the JivaVolume Corresponding to the pvc name
+	jv, err := c.GetJV(pvc.Spec.VolumeName)
+	if err != nil {
+		fmt.Printf("failed to get JivaVolume for %s\n", pvc.Spec.VolumeName)
+		return err
+	}
+	// 2. Fill in Jiva Volume Claim related details
+	jivaPvcInfo := util.JivaPVCInfo{
+		Name:             pvc.Name,
+		Namespace:        pvc.Namespace,
+		CasType:          util.JivaCasType,
+		BoundVolume:      pvc.Spec.VolumeName,
+		StorageClassName: *pvc.Spec.StorageClassName,
+		Size:             util.ConvertToIBytes(jv.Spec.Capacity),
+	}
+	if jv != nil {
```

Comment:
Okay, I forgot to remove the return statement in line 54. 

This check would be there because we would not terminate the command if JV, PV,  etc is not found. The idea is that, we have reached this part of code because a PVC related to this JV exist, so the main aim is to show the PVC describe along with the underlying resources if they exist. If the underlying resources do not exist we would show which resources are missing and fill the details which are available. In the worst case scenario if nothing exists belonging to the PVC, only PVC details would be shown, rest stuffs would be empty and a message that these stuffs are not there would be shown.